src/Symfony/Component/Console/EventListener/ExceptionListener.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Symfony\Component\Console\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleExceptionEvent;
+use Symfony\Component\Console\Event\ConsoleTerminateEvent;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Console exception listener.
+ *
+ * Attempts to log exceptions or abnormal terminations of console commands.
+ *
+ * @author James Halsall <james.t.halsall@googlemail.com>
+ */
+class ExceptionListener implements EventSubscriberInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * Constructor.
+     *
+     * @param LoggerInterface $logger A logger
+     */
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Handles console command exception.
+     *
+     * @param ConsoleExceptionEvent $event Console event
+     */
+    public function onKernelException(ConsoleExceptionEvent $event)
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $exception = $event->getException();
+
+        $this->logger->error($exception->getMessage(), array('exception' => $exception));
```

Comment:
I'm sorry to be very picky, but the log message is not good. I prefer something like that:
`'Exception thrown while running command: "%s". Message: "%s".`

It's better because if the RDBMS goes down (for example), many commands or consumers will fail. So you will receive N time the exact same message. It's better to have N messages because all commands are different and you must know what command fail.